### PR TITLE
Support for uploading and downloading files over gRPC

### PR DESCRIPTION
**Pull request by or-else**

### CODE REVIEW

File: server/hdl_files.go [hunk 1]
```diff
@@ -316,6 +325,228 @@ func largeFileReceive(wrt http.ResponseWriter, req *http.Request) {
...
+		remoteAddr = p.Addr.String()
+	}
+	uid, challenge, err := authFileRequest(authMethod, secret, "", remoteAddr)
+	if err != nil {
+		writeResponse(decodeStoreError(err, msgID, now, nil), err)
+		return nil
+	}
+
+	if challenge != nil {
+		log.Println("3", err)
```
Comment (aforge) [inline server/hdl_files.go:458]: Remove?

File: server/hdl_files.go [hunk 2]
```diff
@@ -316,6 +325,228 @@ func largeFileReceive(wrt http.ResponseWriter, req *http.Request) {
...
+		ObjHeader: types.ObjHeader{
+			Id: store.Store.GetUidString(),
+		},
+		User:     uid.String(),
+		MimeType: mimeType,
+	}
+	fdef.InitTimes()
+
+	reader, writer := io.Pipe()
+	done := make(chan error, 1)
```
Comment (queses) [inline server/hdl_files.go:507]: Can't find where this channel read

**Review by queses — APPROVED** [review]

File: server/hdl_files.go (hunk 1 above)
Comment (or-else) [reply server/hdl_files.go]: Done

File: server/hdl_files.go (hunk 2 above)
Comment (or-else) [reply server/hdl_files.go:506]: Dood catch, thanks. Done.

**Review by aforge — APPROVED** [review]

**Pull request merged**